net/sync/gpoddernet/src/main/java/de/danoeh/antennapod/net/sync/nextcloud/NextcloudSyncService.java

```
@@ -0,0 +1,216 @@
+            String responseString = ResponseMapper.convertStreamToString(inputStream);
+            JSONObject json = new JSONObject(responseString);
+            inputStream.close();
+            return ResponseMapper.readSubscriptionChangesFromJsonObject(json);
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+            throw new IllegalStateException(e);
+        } catch (JSONException | MalformedURLException e) {
+            e.printStackTrace();
+            throw new SyncServiceException(e);
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new SyncServiceException(e);
+        }
+    }
+
+    private HashMap<String, List<String>> getHeaderWithUserAgent() {
+        HashMap<String, List<String>> header = new HashMap<>();
+        header.put("User-Agent", Collections.singletonList(userAgent));
+        return header;
+    }
+
+    @Override
+    public UploadChangesResponse uploadSubscriptionChanges(List<String> addedFeeds, List<String> removedFeeds) {
+        try {
+            HashMap<String, List<String>> header = getHeaderWithUserAgent();
+            header.put("Content-Type", Collections.singletonList("application/json"));
+
+            final JSONObject requestObject = new JSONObject();
+            requestObject.put("add", new JSONArray(addedFeeds));
+            requestObject.put("remove", new JSONArray(removedFeeds));
+
+            NextcloudRequest nextcloudRequest = new NextcloudRequest.Builder()
+                    .setMethod("POST")
+                    .setUrl(Uri.encode("/index.php/apps/gpoddersync/subscription_change/create", "/"))
+                    .setHeader(header)
+                    .setRequestBody(requestObject.toString())
+                    .build();
+
+            this.nextcloudApi.performNetworkRequest(nextcloudRequest).close();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Silently ignoring will never make the service re-try this. Might lead to data not being synced even though the user might think it is.

---

added and throw exception now